### DISCUSSION TITLE
Ensemble with models from different library

### DISCUSSION THREAD

**Discussion by sigord** [alan-turing-institute/sktime#4615] [Q&A] [answered]
Hello everyone!
I am new to time series forecasting and have been given an assignment in this area. Luckily my closed environment included the `gluonts 0.8.1` and `sktime 0.8.0` libraries. I want to build an ensemble in sktime with sktime models, some neural networks from gluonts, xgbregressor from sklearn/xgboost. I am looking for any guidance on how to do this.

Thanks for creating this useful lib.

**Comment by fkiraly** [comment, marked as answer]
hm, `sktime` 0.8.0 is ages ago, are you sure you don´t mean `sktime` 0.18.0?

Now this could be a too old version of `sktime`, but in general you can take the extension templates in order to create `sktime` compatible interfaces for any library's forecaster. The forecaster template is here: https://github.com/sktime/sktime/blob/main/extension_templates/forecasting.py

There is also an in-depth video tutorial, which is also linked in the guide here:
https://www.sktime.net/en/stable/developer_guide/add_estimators.html

`xgbregressor` should work together with `make_reduction`, as it already has an `sklearn` compatible interface, for that you need to use the `sklearn` API: https://xgboost.readthedocs.io/en/stable/python/python_api.html#module-xgboost.sklearn

**Reply by sigord** [reply]
Thanks a lot for the quick reply! Yes, unfortunately I meant 0.8.0. Updating and adding libraries is expensive in some large companies.

**Reply by fkiraly** [reply]
😱

**Reply by fkiraly** [reply]
anyway, if you are up for adding an interface to `gluonts` to `sktime`, that would be appreciated (if you are doing it anyway?)
